pkg/console/subresource/manifestwork/helpers.go

```diff
@@ -0,0 +1,72 @@
+package manifestwork
+
+import (
+	"errors"
+
+	oauthclientsub "github.com/openshift/console-operator/pkg/console/subresource/oauthclient"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+
+	workv1 "open-cluster-management.io/api/work/v1"
+)
+
+// Appends an object to manifests.
+func AppendManifest(manifestWork *workv1.ManifestWork, object runtime.Object) {
+	if &manifestWork.Spec == nil {
+		manifestWork.Spec = workv1.ManifestWorkSpec{}
+	}
+
+	if &manifestWork.Spec.Workload == nil {
+		manifestWork.Spec.Workload = workv1.ManifestsTemplate{}
+	}
+
+	if &manifestWork.Spec.Workload.Manifests == nil {
+		manifestWork.Spec.Workload.Manifests = []workv1.Manifest{}
+	}
+
+	manifestWork.Spec.Workload.Manifests = append(
```

Comment:
Cant we just set the whole `manifestWork.Spec.Workload.Manifests` without those checks? since the stub is returning a object without any spec

Reply:
That's what the checks above are for. We're sequentially checking that all parent properties exist before we get to this line.

Reply:
Right, but based on the API the both `Spec` and `Spec.Workload` are required fields, they for they will be present in the object no matter what.